lib/facter/windows_env.rb

```diff
@@ -0,0 +1,40 @@
+Facter.add('windows_env') do
+  confine osfamily: :windows
+  setcode do
+    require 'puppet/util/windows/process'
+
+    # This list must be in uppercase
+    whitelist = %w[
+      ALLUSERSPROFILE
+      APPDATA
+      COMMONPROGRAMFILES
+      COMMONPROGRAMFILES(X86)
+      HOME
+      HOMEDRIVE
+      HOMEPATH
+      LOCALAPPDATA
+      PATHEXT
+      PROCESSOR_IDENTIFIER
+      PROCESSOR_LEVEL
+      PROCESSOR_REVISION
+      PROGRAMDATA
+      PROGRAMFILES
+      PROGRAMFILES(X86)
+      PSMODULEPATH
+      PUBLIC
+      SYSTEMDRIVE
+      SYSTEMROOT
+      TEMP
+      TMP
+      USERPROFILE
+      WINDIR
+    ]
+
+    result = {}
```

Comment:
Returning an empty hash is equivalent of not showing up as a fact at all correct?

Reply:
Nope....returning an empty hash returns an empty hash e.g.

``` text
    "windows_programfiles": "C:\\Program Files",
    "windows_env": {
    },
    "clientcert": "glenn.sarti-***",
```

Reply:
Okay Thanks for clarifying this 👍